packages/eas-cli/src/submit/ios/__tests__/CredentialsServiceSource-test.ts

```diff
@@ -0,0 +1,94 @@
+import { Platform } from '@expo/eas-build-job';
+import { v4 as uuidv4 } from 'uuid';
+
+import { asMock } from '../../../__tests__/utils';
+import { jester as mockJester } from '../../../credentials/__tests__/fixtures-constants';
+import { testCommonIosAppCredentialsFragment } from '../../../credentials/__tests__/fixtures-ios';
+import { SetUpSubmissionCredentials } from '../../../credentials/ios/actions/SetUpSubmissionCredentials';
+import { createTestProject } from '../../../project/__tests__/project-utils';
+import { getBundleIdentifierAsync } from '../../../project/ios/bundleIdentifier';
+import { createSubmissionContextAsync } from '../../context';
+import { getFromCredentialsServiceAsync } from '../CredentialsServiceSource';
+
+jest.mock('fs');
+jest.mock('../../../prompts');
+jest.mock('../../../credentials/utils/promptForCredentials');
+jest.mock('../../../user/User', () => ({
+  getUserAsync: jest.fn(() => mockJester),
+}));
+jest.mock('../../../user/Account', () => ({
+  findAccountByName: jest.fn(() => mockJester.accounts[0]),
+}));
+jest.mock('../../../project/ios/bundleIdentifier');
+
+const testProject = createTestProject(mockJester, {
+  android: {
+    package: 'com.expo.test.project',
+  },
+});
+const mockManifest = { exp: testProject.appJSON.expo };
+jest.mock('@expo/config', () => ({
+  getConfig: jest.fn(() => mockManifest),
+}));
+const projectId = uuidv4();
+
+beforeEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe(getFromCredentialsServiceAsync, () => {
+  it('returns an App Specific Password from the credentialService source', async () => {
+    const ctx = await createSubmissionContextAsync({
+      platform: Platform.IOS,
+      projectDir: testProject.projectRoot,
+      projectId,
+      archiveFlags: {
+        url: 'http://expo.dev/fake.apk',
+      },
+      profile: {
+        language: 'en-US',
+      },
+      nonInteractive: true,
+    });
+    jest
+      .spyOn(SetUpSubmissionCredentials.prototype, 'runAsync')
+      .mockImplementation(async _ctx => 'super secret');
+    asMock(getBundleIdentifierAsync).mockImplementation(() => 'com.hello.world');
+    const result = await getFromCredentialsServiceAsync(ctx);
```

Comment:
```suggestion

    const result = await getFromCredentialsServiceAsync(ctx);
```
nitpick: IMO one blank line between preparation and act/assertion part makes tests easier to read, but up to you.